File: 8.cs
using System;

class Program
{
    static void Main()
    {
        decimal input = 1;
        decimal money = 1000 - input;

        if (input == 1000)
        {
            Console.WriteLine("ไม่มีเงินทอน");
        }
        else if (input > 1000)
        {
            Console.WriteLine("Error");
        }
        else
        {
            if (money >= 1000)
            {
                int sum1 = (int)(money / 1000);
                int sum1s = sum1 * 1000;
                money -= sum1s;
                Console.WriteLine($"ธนบัตร 1,000 = {sum1} ใบ");
            }
            if (money >= 500)
            {
                int sum2 = (int)(money / 500);
                int sum2s = sum2 * 500;
                money -= sum2s;
                Console.WriteLine($"ธนบัตร 500 = {sum2} ใบ");
            }
            if (money >= 100)
            {
                int sum3 = (int)(money / 100);
                int sum3s = sum3 * 100;
                money -= sum3s;
                Console.WriteLine($"ธนบัตร 100 = {sum3} ใบ");
            }
            if (money >= 50)
            {
                int sum4 = (int)(money / 50);
                int sum4s = sum4 * 50;
                money -= sum4s;
                Console.WriteLine($"ธนบัตร 50 = {sum4} ใบ");
            }
            if (money >= 20)
            {
                int sum5 = (int)(money / 20);
                int sum5s = sum5 * 20;
                money -= sum5s;
                Console.WriteLine($"ธนบัตร 20 = {sum5} ใบ");
            }
            if (money >= 10)
            {
                int sum51 = (int)(money / 10);
                int sum51s = sum51 * 10;
                money -= sum51s;
                Console.WriteLine($"เหรียญ 10 = {sum51} เหรียญ");
            }
            if (money >= 5)
            {
                int sum6 = (int)(money / 5);
                int sum6s = sum6 * 5;
                money -= sum6s;
                Console.WriteLine($"เหรียญ 5 = {sum6} เหรียญ");
            }
            if (money >= 2)
            {
                int sum7 = (int)(money / 2);
                int sum7s = sum7 * 2;
                money -= sum7s;
                Console.WriteLine($"เหรียญ 2 = {sum7} เหรียญ");
            }
            if (money >= 1)
            {
                int sum8 = (int)money;
                int sum8s = sum8;
                money -= sum8s;
                Console.WriteLine($"เหรียญ 1 = {sum8} เหรียญ");
            }
            if (money >= 0.75m)
            {
                Console.WriteLine("เหรียญ 0.75 = 1 เหรียญ");
                money -= 0.75m;
            }
            if (money >= 0.50m)
            {
                Console.WriteLine("เหรียญ 0.50 = 1 เหรียญ");
                money -= 0.50m;
            }
        }
    }
}

File: 8.php
<?php
$input =1;
$money=1000-$input;
	if($input==1000){
    echo "ไม่มีเงินทอน <br>";
    }
	if($input>1000){
    echo "Error";
    }
  if($money>=1000){
      $sum1=$money/1000;
      $sum1s=floor($sum1)*1000;
      $money=$money-$sum1s;
      echo "ธนบัตร 1,000 = ".floor($sum1)." ใบ"."<br>";
  }if($money>=500){
      $sum2=$money/500;
      $sum2s=floor($sum2)*500;
      $money=$money-$sum2s;
      echo "ธนบัตร 500 = ".floor($sum2)." ใบ"."<br>";
  }if($money>=100){
      $sum3=$money/100;
      $sum3s=floor($sum3)*100;
      $money=$money-$sum3s;
      echo "ธนบัตร 100 = ".floor($sum3)." ใบ"."<br>";
  }if($money>=50){
      $sum4=$money/50;
      $sum4s=floor($sum4)*50;
      $money=$money-$sum4s;
      echo "ธนบัตร 50 = ".floor($sum4)." ใบ"."<br>";
  }if($money>=20){
      $sum5=$money/20;
      $sum5s=floor($sum5)*20;
      $money=$money-$sum5s;
      echo "ธนบัตร 20 = ".floor($sum5)." ใบ"."<br>";
  }if($money>=10){
      $sum51=$money/10;
      $sum51s=floor($sum51)*10;
      $money=$money-$sum51s;
      echo "เหรียญ 10 = ".floor($sum51)." เหรียญ"."<br>";
  }if($money>=5){
      $sum6=$money/5;
      $sum6s=floor($sum6)*5;
      $money=$money-$sum6s;
      echo "เหรียญ 5 = ".floor($sum6)." เหรียญ"."<br>";
  }if($money>=2){
      $sum7=$money/2;
      $sum7s=floor($sum7)*2;
      $money=$money-$sum7s;
      echo "เหรียญ 2 = ".floor($sum7)." เหรียญ"."<br>";
  }if($money>=1){
      $sum8=$money/1;
      $sum8s=floor($sum8)*1;
      $money=$money-$sum8s;
      echo "เหรียญ 1 = ".floor($sum8)." เหรียญ"."<br>";
  }if($money>=0.75){            
      echo "เหรียญ 0.75 = 1 เหรียญ"."<br>";
      $money=$money-0.75;  
  }if($money>=0.50){
      $money=$money-0.50;      
      echo "เหรียญ 0.50 = 1 เหรียญ"."<br>";
  }
  ?>
